### ISSUE TITLE
Cannot set property 'getClientExtensionResults' of null

### ISSUE THREAD

**Issue by tarun14110** [chrome-key/ckey#5]
Getting error `Cannot set property 'getClientExtensionResults' of null` while trying to use this extension at [https://demo.yubico.com/](https://demo.yubico.com/)

On looking at the error in the extension code in chrome. I found the below error. Can you please help me out here?
![Screenshot from 2020-01-29 17-11-02](https://user-images.githubusercontent.com/12219700/73408738-760af000-42ba-11ea-8fc2-061d5fe8b36a.png)

**Comment by Hartimer** [comment]
Thanks for the feedback @tarun14110. There's definitely some room for improvement on that error message.

**TL;DR**
This extension does not support direct attestation ([spec details](https://www.w3.org/TR/webauthn/#dom-attestationconveyancepreference-direct))

**Long answer**

The direct attestation requires the device itself to provide a certificate that confirms that it is an acceptable one.
Yubico yubikeys are embedded with a certificate at the time of manufacturing, so you can verify (and limit) the manufacturers you which to support, by requiring them to be under a certain CA. What that page is likely doing is precisely checking the device certificate to match against whichever model you wish to test. We cannot satisfy that request because they have not issued us a certificate from said CA (nor will they ever).

This extension is for development purposes only, it is not feasible to provide a CA as (1) it would not be very secure (2) it would be highly unlikely that it would be recognized as a valid manufacturer certificate.

As such, we can only support no attestation or potentially indirect.

**Comment by tarun14110** [comment]
Thanks, @Hartimer. That was helpful. I am able to register and authenticate after setting the `attestation` field as `none`.

**Comment by Hartimer** [comment]
@tarun14110 about that last question, that is a known limitation of the extension. I haven't wired up the cancellation of the other popups yet. Annoying at times I know.

I'm glad to hear it worked. I'm closing this issue for now since it is working as intended.
